Node pools: add info on autoscaling behaviour (#387)
giantswarm/docs: src/content/basics/nodepools/index.md

The new definition schema for v5 allows for defining cluster and node pool details in one file,
to be submitted for creation via the [`gsctl create cluster`](/reference/gsctl/create-cluster/) command.

## Node pools and autoscaling {#autoscaling}

> With node pools, you set the autoscaling range per node pool. The Kubernetes cluster autoscaler has to decide which node pool to scale under which circumstances.

Reviewer: ```suggestion
With node pools, the autoscaling is configured per node pool automatically. The Kubernetes cluster autoscaler has to decide which node pool to scale under which circumstances.
```
In reality customer does not perform any action

Author: Actually, yes. The customer initially decides in which range the node pool can scale.

Reviewer: in which range? can you explain it a bit more? what do you mean by range?

Author: I mean the number of nodes, which is configured by the user as a MIN and MAX number.

Reviewer: AH ok I was thinking from the ASG configuration side

> If you assign workloads to node pools as described [above](#assigning-workloads) and the autoscaler finds pods that are not assigned to any node, it will decide based on the node selectors which node pools to scale up.

Reviewer: ```suggestion
If you assign workloads to node pools as described [above](#assigning-workloads) and the autoscaler finds pods in `Pending` state, it will decide based on the node selectors which node pools to scale up.
```

> In case there are workloads not assiged to any node pools, the autoscaler may pick any node pool for scaling. For details on the decision logic, pelaase check the upstream [FAQ for AWS](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md).

Reviewer: ```suggestion
In case there are workloads not assigned to any node pools, the autoscaler may pick any node pool for scaling. For details on the decision logic, please check the upstream [FAQ for AWS](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md).
```
